src/common/datasource/src/error.rs

```diff
@@ -89,6 +91,30 @@ pub enum Error {
         location: Location,
         source: tokio::task::JoinError,
     },
+
+    #[snafu(display("Failed to parse delimiter: {}", source))]
+    ParseDelimiter {
+        source: ParseIntError,
+        location: Location,
+    },
+
+    #[snafu(display("Failed to parse shcema infer max record: {}", source))]
+    ParseSchemaInferMaxRecord {
+        source: ParseIntError,
+        location: Location,
+    },
+
+    #[snafu(display("Failed to parse has header: {}", source))]
+    ParseHasHeader {
+        source: ParseBoolError,
+        location: Location,
+    },
+
+    #[snafu(display("Failed to merge schema: {}", source))]
+    MergeSchema {
+        source: arrow_schema::ArrowError,
+        location: Location,
+    },
```

Comment:
Maybe we don't have to define a error variant for every key. Just a single 

```rust
ParseFormat {
    key: String,
    value: String,
    location: Location,
}
```
And map error like:
```rust
    CompressionType::from_str(compression_type).map_err(|_| {
        ParseFormatSnafu {
            key: file_format::FORMAT_COMPRESSION_TYPE,
             value: compression_type,
        }.build()
    }?
)
```
Would be enough

Reply:
If the key is always a constant, consider `&'static str` for the key type.